### PR TITLE
Ogr migration

### LINKED ISSUE #250: replace IGitt with ogr
**Is your feature request related to a problem? Please describe.**
n/a

**Describe the solution you'd like**
IGitt should be migrated to https://github.com/packit-service/ogr

**Describe alternatives you've considered**
a) keep the constantly outdating IGitt
b) invest in keeping IGitt up to date

**Additional context**
problems with IGitt's dependencies 
https://github.com/AICoE/sesheta/issues/38

Related: #206

### PR DESCRIPTION
**Pull request by saisankargochhayat**
- [x] Need to remove object references from internal managers.

- [x] Need to add add rest methods to assign users to issues.

- [x] Replace `source_management.py` with `source_management_ogr`

- [x] Remove `source_management_old.py`, remove IGitt from pipfile.

- [x] Gitlab assign works with usernames.

- [ ] Update documentation.

Closes #250

### CODE REVIEW

File: kebechet/source_management_ogr.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+# Kebechet
+# Copyright(C) 2018, 2019, 2020 Fridolin Pokorny
```
Comment (KPostOffice) [inline kebechet/source_management_ogr.py]: Copy pasta

File: kebechet/source_management_ogr.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                repo=self.repo, namespace=self.namespace
+            )
+        else:
+            raise NotImplementedError
+
+    def get_issue(self, title: str) -> Issue:
+        """Retrieve issue with the given title."""
+        for issue in self.repository.get_issue_list():
+            if issue._raw_issue.title == title:
+                return issue
```
Comment (KPostOffice) [inline kebechet/source_management_ogr.py]: Checkout https://github.com/thoth-station/solver/issues/270 and https://github.com/thoth-station/solver/issues/243 , not sure what the expected behavior would be here.  Should we get the oldest one, the newest one, or both?

**Comment by saisankargochhayat** [comment]
@fridex I have replace source_management with ogr and also migrated the managers to use only ogr objects. Let me know if we are missing out on anything else?

File: kebechet/source_management_ogr.py (hunk 2 above)
Comment (fridex) [reply kebechet/source_management_ogr.py]: The newest one.

**Comment by fridex** [comment]
> @fridex I have replace source_management with ogr and also migrated the managers to use only ogr objects. Let me know if we are missing out on anything else?

Cool, this looks nice! Please go ahead and replace the old one with you implementation so its visible to reviewer what changes you have made.

You can also remove igitt from Pipfile/Pipfile.lock and requirements.txt.

File: kebechet/source_management_old.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        issue = self.get_issue(title)
+        if issue:
+            _LOGGER.info(f"Issue already noted on upstream with id #{issue.number}")
+            if not refresh_comment:
+                return issue
+
+            comment_body = refresh_comment(issue)
+            if comment_body:
+                issue.add_comment(comment_body)
+                _LOGGER.info(f"Added refresh comment to issue #{issue.number}")
```
Comment (fridex) [inline kebechet/source_management_old.py]: It's better to use lazy debug string evaluation:

```
                _LOGGER.info("Added refresh comment to issue #issue.number", issue.number)
```

File: kebechet/source_management.py [hunk 4]
```diff
@@ -100,90 +114,65 @@ def close_issue_if_exists(self, title: str, comment: str = None):
...
+        try:
+            merge_request = self.repository.create_pr(
+                commit_msg, body, "master", branch_name
+            )
+            merge_request.add_label(labels)
 
-        merge_request.labels = set(labels or [])
-        return merge_request
+        except Exception as exc:
+            raise RuntimeError(f"Failed to create a pull request: {exc}")
```
Comment (fridex) [inline kebechet/source_management.py]: It's not been done in the previous implementation - in general, its a good practice to create project specific exceptions so its clear what errors are raised within the project and to the user.

File: kebechet/source_management_ogr.py (hunk 2 above)
Comment (saisankargochhayat) [reply kebechet/source_management_ogr.py]: This should get the newest one :+1:

File: requirements.txt [hunk 5]
```diff
@@ -4,8 +4,6 @@ pipenv
 toml
 delegator.py
 pyyaml
-IGitt==0.4.1.dev20180722120744
-# Pinned due to igitt pinning.
```
Comment (fridex) [inline requirements.txt]: ogr is missing here.

File: kebechet/exception.py [hunk 6]
```diff
@@ -62,3 +62,27 @@ class WebhookPayloadError(KebechetException):
 
     Generally the payload is empty or not of proper json format.
     """
+
+
+class CannotFetchPRError(KebechetException):
+    """
+    An exception raised if there OGR cannot fetch the PR's for a repo.
```
Comment (fridex) [inline kebechet/exception.py]: Docstring starts with a single sentence, coala/linters will argue here:

```python
    """An exception raised if there OGR cannot fetch the PR's for a repo.

    Generally the token is invalid or doesn't have access to the repo.
    """
```

Here and bellow

File: requirements.txt (hunk 5 above)
Comment (saisankargochhayat) [reply requirements.txt]: I though `requirements.txt` is obsolete :stuck_out_tongue: I will add it now!

**Comment by goern** [comment]
@pacospace can we trace back why this check is hanging for 12hrs?!

**Pull request merged**